otel-tests/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use quote::ToTokens;
+use syn::ItemFn;
+use syn::Lit;
+use syn::MetaNameValue;
+use syn::Path;
+
+#[proc_macro_attribute]
+pub fn otel_test(attribute: TokenStream, item: TokenStream) -> TokenStream {
+    let fn_item = syn::parse::<ItemFn>(item).unwrap();
+
+    let sig = fn_item.sig;
+    let block = fn_item.block;
+    let name = &sig.ident;
+    let attrs = fn_item.attrs;
+
+    let test = if attrs.is_empty() {
+        quote!(#[tokio::test])
+    } else {
+        quote!()
+    };
+
+    let init_tracing = if attribute.is_empty() {
+        quote!(::daemon_tests::init_tracing)
+    } else {
+        let attribute = syn::parse::<MetaNameValue>(attribute).unwrap();
+        if !attribute.path.is_ident("init") {
+            panic!(
+                "Unexpected attribute {:?}",
+                attribute.path.get_ident().unwrap()
+            )
+        }
+
+        let path = match attribute.lit {
+            Lit::Str(lit) => lit.parse::<Path>().expect("Expected function path"),
+            other => panic!(
+                "Expected string function path, got {}",
+                other.to_token_stream()
+            ),
+        };
+
+        quote!(#path)
+    };
+
+    let tokens = quote! {
+        #test
+        #(#attrs)*
+        #sig {
+            #init_tracing();
+
+            let caught = {
+                futures::FutureExt::catch_unwind(std::panic::AssertUnwindSafe(tracing::Instrument::instrument(
+                    async #block, tracing::info_span!(stringify!(#name))
+                ))).await
+            };
+
+            #[allow(clippy::disallowed_methods)]
+            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
```

Comment:
why do we need a sleep here?

Reply:
I'm not entirely sure :( It makes it all work, and I theorise that it lets the otel thread receive the spans, and then when we flush they are there to be flushed. If it's removed, the flush doesn't as consistently reduce spans being dropped.